substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -291,10 +314,13 @@ where
 		}
 
 		let signature = is_signed.then(|| Decode::decode(input)).transpose()?;
-		let function = Decode::decode(input)?;
 
-		if let Some((before_length, after_length)) =
-			input.remaining_len()?.and_then(|a| before_length.map(|b| (b, a)))
+		let mut input = CopyReader::new(input);
+		// We have to check that the function decodes correctly.
+		Call::decode(&mut input)?;
```

Comment:
We really finally need to get some proper `skip` implemented.

Reply:
If you mean to just skip the encoded function bytes without decoding it, the problem is that we will have to return a `Result<>` from the methods that decode the function and:
1. I'm not sure how hard it will be to propagate this `Result<>` to all the call sites 
2. I'm not sure if it's ok to panic at a later time for example if the function can't be decoded

I will look into it

Reply:
Ok, I think I finally got it. I replaced this with `Call::skip(&mut input)`, but from what I understand it still does `decode` internally. I agree we should implement a proper skip, but maybe in a separate PR ? I guess it's something that we should do in the `parity-scale-codec` crate, right ?